lib/ui/painting.dart

```diff
@@ -1606,22 +1606,30 @@ class Codec extends NativeFieldWrapperClass2 {
 /// Instantiates an image codec [Codec] object.
 ///
 /// [list] is the binary image data (e.g a PNG or GIF binary data).
-/// The data can be for either static or animated images.
+/// The data can be for either static or animated images. The following image
+/// formats are supported: {@macro flutter.dart:ui.imageFormats}
 ///
-/// The following image formats are supported: {@macro flutter.dart:ui.imageFormats}
+/// [decodedCacheRatioCap] is the default maximum multiple of the compressed
+/// image size to cache when decoding animated image frames. For example,
+/// setting this to `2.0` means that a 400KB GIF would be allowed at most to use
+/// 800KB of memory caching unessential decoded frames. Caching decoded frames
+/// saves CPU but can result in out-of-memory crashes when decoding large (or
+/// multiple) animated images.
 ///
 /// The returned future can complete with an error if the image decoding has
 /// failed.
-Future<Codec> instantiateImageCodec(Uint8List list) {
+Future<Codec> instantiateImageCodec(Uint8List list,
+                                   {double decodedCacheRatioCap = double.maxFinite}) {
   return _futurize(
-    (_Callback<Codec> callback) => _instantiateImageCodec(list, callback, null)
+    (_Callback<Codec> callback) =>
+        _instantiateImageCodec(list, callback, null, decodedCacheRatioCap)
   );
 }
```

Comment:
Please use flutter style, which would make this:

```dart
Future<Codec> instantiateImageCodec(Uint8List list, {
  double decodedCacheRatioCap = double.maxFinite,
}) {
  return _futurize(
    (_Callback<Codec> callback) => _instantiateImageCodec(list, callback, null, decodedCacheRatioCap),
  );
}
```

or if you really don't want a slightly long line:

```dart
Future<Codec> instantiateImageCodec(Uint8List list, {
  double decodedCacheRatioCap = double.maxFinite,
}) {
  return _futurize((_Callback<Codec> callback) {
    return _instantiateImageCodec(list, callback, null, decodedCacheRatioCap);
  });
}
```

...or if you for some reason need the shortest possible lines:

```dart
Future<Codec> instantiateImageCodec(Uint8List list, {
  double decodedCacheRatioCap = double.maxFinite,
}) {
  return _futurize(
    (_Callback<Codec> callback) {
      return _instantiateImageCodec(
        list,
        callback,
        null,
        decodedCacheRatioCap,
      );
    },
  );
}
```